pkg/catalog/outbound_traffic_policies.go

```diff
@@ -43,15 +49,29 @@ func (mc *MeshCatalog) listOutboundPoliciesForTrafficTargets(downstreamIdentity
 		if !isValidTrafficTarget(t) {
 			continue
 		}
+		var retryPolicy trafficpolicy.RetryPolicy
+		retryPolicyHasReqField := 0
+		addRetryPolicy := false
+		for annotationKey, annotationVal := range t.Annotations {
+			addRetryPolicy, retryPolicyHasReqField = getRetryPolicy(retryPolicy, annotationKey, annotationVal)
+		}
+		if retryPolicyHasReqField != 2 {
+			addRetryPolicy = false
+			log.Error().Msgf("Retry policy does not have all required fields: %v, %v", constants.EnableRetryKey, constants.RetryOnKey)
+		}
```

Comment:
I suggest wrapping this within a feature flag similar to `enableIngressBackendPolicy`.

Reply:
Also let's simplify this logic here to:
```
retryPolicy, err = getRetryPolicyForTrafficTarget(t)
...
```

`getRetryPolicyForTrafficTarget()` should parse the annotations on the traffic target and return (trafficpolicy.RetryPolicy, error).

Reply:
PR to add feature flag for retry policy: https://github.com/openservicemesh/osm/pull/4029